File: app/src/main/java/com/thales/github/ui/pullresquest/PullRequestContract.java
package com.thales.github.ui.pullresquest;

import com.thales.github.model.pullrequest.PullRequest;
import com.thales.github.ui.BasePresenter;

import java.util.List;

interface PullRequestContract {
    interface View {
        void showPullRequests(List<PullRequest> pullRequests);

        void showProgress();

        void hideProgress();

        void showSnackbarNoConnection();

        void showSnackbarError();

        void showNoRepositoriesMessage();

        void hideNoRepositoriesMessage();

        boolean isNetworkAvailable();
    }

    interface Presenter extends BasePresenter {
        void loadPullRequests(String title, String username);
    }
}

File: app/src/main/java/com/thales/github/ui/pullresquest/PullRequestContract.kt
package com.thales.github.ui.pullresquest

import com.thales.github.model.pullrequest.PullRequest
import com.thales.github.ui.BasePresenter

internal interface PullRequestContract {
    interface View {
        fun showPullRequests(pullRequests: List<PullRequest>)

        fun showProgress()

        fun hideProgress()

        fun showSnackbarNoConnection()

        fun showSnackbarError()

        fun showNoRepositoriesMessage()

        fun hideNoRepositoriesMessage()

        fun isNetworkAvailable(): Boolean
    }

    interface Presenter : BasePresenter {
        fun loadPullRequests(title: String, username: String)
    }
}
